napari/layers/image/image.py

```diff
@@ -354,15 +366,10 @@ def __init__(
         # triggered (self._update_dims(), below).
         self._set_colormap(colormap)
         self.contrast_limits = self._contrast_limits
-        self._interpolation = {
-            2: Interpolation.NEAREST,
-            3: (
-                Interpolation3D.NEAREST
-                if self.__class__.__name__ == 'Labels'
-                else Interpolation3D.LINEAR
-            ),
-        }
-        self.interpolation = interpolation
+        self._interpolation2d = Interpolation.NEAREST
+        self._interpolation3d = Interpolation3D.NEAREST
```

Comment:
Why do you need to set the private attributes here? Won't the setters called immediately below take care of that?

Reply:
there is strong advise to not define attribute outside constructor.

Reply:
Oh right. I think that I would personally break that rule here because it's pretty clear what's going on (i.e. we're using the corresponding property setters and not some other method).

But I don't feel strongly, so feel free to leave this as is.